BookReader/BookReader.js

```diff
@@ -1430,13 +1418,26 @@ BookReader.prototype.nextReduce = function( currentReduce, direction, reductionF
             }
         }
         return reductionFactors[newReduceIndex];
-    }
-
-    // Asked for specific autofit mode
-    for (var i = 0; i < reductionFactors.length; i++) {
-        if (reductionFactors[i].autofit == direction) {
-            return reductionFactors[i];
-        }
+    } else if (direction === 'auto') {
+      // Auto mode chooses the least reduction
+      var choice = null;
```

Comment:
Suggestion: Move this declaration (not the assignment, obviously) up to the top of the function to prevent hoisting confusion.

Reply:
Thus far, the BookReader.js file does not deal with this, so I will leave it as is.